h2o-core/src/main/java/water/Scope.java

```diff
@@ -92,4 +89,7 @@ static public void untrack(Iterable<Key<Vec>> keys) {
     HashSet<Key> xkeys = scope._keys.peek();
     for (Key<Vec> key : keys) xkeys.remove(key); // Untrack key
   }
+
+  @Override
```

Comment:
You are right. It is confusing. I can make the public constructor into package-visible.
But then what? We are still in a situation where Scope is being used the way it is being used.
We either stop using static and thread-local, and do everything the modern way, with try/resource, or keep it the way it is, and keep the new way as an opportunity, a road to follow, or just change all the use cases and switch to the new way. I find this new way correct.

I would love to discuss it all, but currently I'm not in a position to openly discuss all this. Let's stop somewhere. What are your specific suggestions, Michal? Pasha, what's your opinion? As I understand, after 3 weeks of discussions there's no rush at all. I'll be happy to make it work either way.

If you want to set up a vote, there should be at least an odd number of voters, I believe.